routes/transaction.go

```diff
@@ -69,21 +69,42 @@ func (fes *APIServer) GetTxn(ww http.ResponseWriter, req *http.Request) {
 		copy(txnHash[:], txnHashBytes)
 	}
 
-	txnFound := false
+	// The order of operations is tricky here. We need to do the following in this
+	// exact order:
+	// 1. Check the mempool for the txn
+	// 2. Wait for txindex to fully sync
+	// 3. Then check txindex
+	//
+	// If we instead check the mempool afterward, then there is a chance that the txn
+	// has been removed by a new block that is not yet in txindex. This would cause the
+	// endpoint to incorrectly report that the txn doesn't exist on the node, when in
+	// fact it is in "limbo" between the mempool and txindex.
 	txnStatus := requestData.TxnStatus
 	if txnStatus == "" {
 		txnStatus = TxnStatusInMempool
 	}
+	txnInMempool := fes.backendServer.GetMempool().IsTransactionInPool(txnHash)
+	startTime := time.Now()
+	// We have to wait until txindex has reached the uncommitted tip height, not the
+	// committed tip height. Otherwise we'll be missing ~2 blocks in limbo.
+	coreChainTipHeight := fes.TXIndex.CoreChain.BlockTip().Height
+	for fes.TXIndex.TXIndexChain.BlockTip().Height < coreChainTipHeight {
+		if time.Since(startTime) > 30*time.Second {
```

Comment:
Can we lower this value?

Reply:
It's a maximum value, and it will always always return faster than this. That said, it can sometimes take a long time for txindex to catch up and I think it's good to leave it at this value.